gateway.go

```diff
@@ -36,6 +48,20 @@ func (gateway *Gateway) ListenAndServe(proxies []*Proxy) error {
 	return nil
 }
 
+func (gateway *Gateway) EnablePrometheus(bind string) error {
+	gateway.wg.Add(1)
```

Comment:
Could you elaborate the benefit or reason of doing this?

Reply:
At the moment, once a proxy config is removed, players are not actually disconnected. Their goroutine stays active in the background piping their connection. In addition to that, the listening goroutine becomes stuck in `.Listen()`, because the listener has been closed, causing the port to be unreachable causing `.Listen()` to never return (at least in my tests). Nonetheless, I saw what you were going for here, to keep the program running as long as there are active proxies. Because of what I mentioned above about players not being disconnected when a proxy is removed, neither should they actually be disconnected when the last proxy is removed, unless the process is explicitly stopped. So I decided to add the Prometheus endpoint to the waitgroup, because it should keep the process active to expose the counts of those players still connected.

This might seem like a weird reasoning, since it's basically because of 2 bugs combined, but I wasn't sure what the design choices were regarding the removal of a proxy, so I decided to choose the safe route.

Reply:
dont worry its not a weird reason at all. I was thinking of the same reason but just wanted to check with you

Reply:
@SnowpMakes good point. I never thought of that. I will definitely have to refactor Infrared because it will be difficult to put Bedrock support in there right now. Even supporting multiple Minecraft versions feels kinda hacky at the moment.